plugins/outputs/azure_data_explorer/azure_data_explorer_test.go

```diff
@@ -0,0 +1,152 @@
+	"github.com/Azure/azure-kusto-go/kusto/ingest"
+	"github.com/influxdata/telegraf"
+	telegrafJson "github.com/influxdata/telegraf/plugins/serializers/json"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+const createTableCommandExpected = `.create-merge table ['%s']  (['fields']:dynamic, ['name']:string, ['tags']:dynamic, ['timestamp']:datetime);`
+const createTableMappingCommandExpected = `.create-or-alter table ['%s'] ingestion json mapping '%s_mapping' '[{"column":"fields", "Properties":{"Path":"$[\'fields\']"}},{"column":"name", "Properties":{"Path":"$[\'name\']"}},{"column":"tags", "Properties":{"Path":"$[\'tags\']"}},{"column":"timestamp", "Properties":{"Path":"$[\'timestamp\']"}}]'`
+
+func TestWrite(t *testing.T) {
+	testCases := []struct {
+		name           string
+		inputMetric    []telegraf.Metric
+		client         *fakeClient
+		createIngestor ingestorFactory
+		expected       map[string]interface{}
+	}{
+		{
+			name:        "Valid metric",
+			inputMetric: testutil.MockMetrics(),
+			client: &fakeClient{
+				queries: make([]string, 0),
+				internalMgmt: func(f *fakeClient, ctx context.Context, db string, query kusto.Stmt, options ...kusto.MgmtOption) (*kusto.RowIterator, error) {
+					f.queries = append(f.queries, query.String())
+					return &kusto.RowIterator{}, nil
+				},
+			},
+			createIngestor: createFakeIngestor,
+			expected: map[string]interface{}{
+				"metricName":                "test1",
+				"createTableCommand":        "",
+				"createTableMappingCommand": "",
+				"error":                     "",
+			},
+		},
+		{
+			name:        "Error in Mgmt",
+			inputMetric: testutil.MockMetrics(),
+			client: &fakeClient{
+				queries: make([]string, 0),
+				internalMgmt: func(f *fakeClient, ctx context.Context, db string, query kusto.Stmt, options ...kusto.MgmtOption) (*kusto.RowIterator, error) {
+					return nil, errors.New("Something went wrong")
+				},
+			},
+			createIngestor: createFakeIngestor,
+			expected: map[string]interface{}{
+				"metricName":                "test1",
+				"createTableCommand":        "",
+				"createTableMappingCommand": "",
+				"error":                     "Something went wrong",
+			},
+		},
+	}
+
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			serializer, _ := telegrafJson.NewSerializer(time.Second)
+
+			plugin := AzureDataExplorer{
+				Endpoint:       "someendpoint",
+				Database:       "databasename",
+				ClientID:       "longclientid",
+				ClientSecret:   "longclientsecret",
+				TenantID:       "longtenantid",
+				Log:            testutil.Logger{},
+				client:         tC.client,
+				ingesters:      map[string]localIngestor{},
+				createIngestor: tC.createIngestor,
+				serializer:     serializer,
+			}
+
+			errorInWrite := plugin.Write(testutil.MockMetrics())
+
+			if tC.expected["error"] != "" {
+				require.EqualError(t, errorInWrite, tC.expected["error"].(string))
```

Comment:
You might want to move `error` to an own `string` field in your test-table to avoid the type assertion and simplify code.